worker/draft.go

```diff
@@ -1839,7 +1839,7 @@ func (n *node) InitAndStartNode() {
 			// snapshot as needed after joining the group, instead of us forcing one upfront.
 			glog.Infoln("Trying to join peers.")
 			n.retryUntilSuccess(n.joinPeers, time.Second)
-			n.SetRaft(raft.StartNode(n.Cfg, nil))
+			n.SetRaft(raft.RestartNode(n.Cfg))
```

Comment:
@mangalaman93 I think we should take a closer look here because this case is not of `restart`. It is possible that the API is same and start/restart is just syntactic sugar but this is counter intuitive. At the very least we should add a comment on top of this as well.

Reply:
Done

Reply:
Tests are not happy when I switch to using StartNode. Let me find an explanation for it.

Reply:
This is resolved?

Reply:
Yes, you should see a comment in the diff now.